examples/simple-non-wasm/src/main.rs

```diff
@@ -0,0 +1,217 @@
+pub struct AnotherCountEvent {
+    count: u64,
+    timestamp: u64,
+    address: Address,
+}
+
+fn another_count_handler(data: Vec<Vec<u8>>) -> Result<Option<EntityResult>> {
+    info!("another_count_handler invoked");
+    if let Some(data) = data.first() {
+        // let tokens: Vec<Token> = Receipt::LogData::into_tokens(data)?;
+        let tokens = ABIDecoder::new()
+            .decode(
+                &[
+                    ParamType::B256,
+                    ParamType::U64,
+                    ParamType::U64,
+                    ParamType::U64,
+                    ParamType::U64,
+                    ParamType::B256,
+                    ParamType::Array(Box::new(ParamType::U8), 16),
+                    ParamType::U64,
+                    ParamType::U64,
+                ],
+                data,
+            )
+            .expect("Bad Encoding!");
+
+        let event = CountEvent::from_token(tokens[6].to_owned()).unwrap();
+
+        info!("another_count_handler received event: {:?}", event);
+    }
+
+    Ok(None)
+}
+
+#[tokio::main]
+pub async fn main() -> Result<()> {
+    // Uneccessary, but helpful
+    let filter = match std::env::var_os("RUST_LOG") {
+        Some(_) => EnvFilter::try_from_default_env().expect("Invalid `RUST_LOG` provided"),
+        None => EnvFilter::new("info"),
+    };
+
+    tracing_subscriber::fmt::Subscriber::builder()
+        .with_writer(std::io::stderr)
+        .with_env_filter(filter)
+        .init();
+
+    let opt = Args::from_args();
+
+    // Load node config
+    let config: IndexerConfig = IndexerConfig::from_file(&opt.config).await?;
+
+    // Load executor config
+    let manifest = Manifest::new(
+        "simple_handler".to_owned(),
+        "schema/counter.graphql".to_owned(),
+        None,
+    );
+
+    let schema = manifest.graphql_schema().await?;
+
+    // In this example, we've already started our fuel node on another process
+    info!("Fuel node listening on {}", config.fuel_node_addr);
+    let api_handle = tokio::spawn(GraphQlApi::run(config.clone()));
+
+    // Create a new service to run
+    let mut service = IndexerService::new(config.clone())?;
+    let _ = service.build_schema(&manifest, &schema)?;
+
+    // Create a new executor to run on the service
+    let mut executor = CustomIndexExecutor::new(&config.database_url, manifest.clone())?;
+
+    // Add some handlers to the executor, in order to process events
+    executor
+        .handler(CustomHandler::new(
+            ReceiptEvent::ReturnData,
+            manifest.namespace.clone(),
+            "Count".to_owned(),
+            count_handler,
+        ))
+        .handler(CustomHandler::new(
+            ReceiptEvent::LogData,
+            manifest.namespace.clone(),
+            "AnotherCount".to_owned(),
+            another_count_handler,
+        ));
+
+    // Add the executor to our service
+    if let Err(e) = service.add_executor(executor, "an_unused_field", manifest, false) {
```

Comment:
You'll notice throughout I left fields untouched, such as `"an_unused_field"` in an attempt to change as little of the existing API as possible